Show unstashable tabs in list when Show All Open Tabs option is enabled.

KerfuffleV2 opened the pull request:
This pull allows unstashable tabs to show up in the "Open Tabs" list. The action icon to stash an unstashable tab will be hidden while it's in an unstashable state. I'm not sure if more protection against stashing the tab than this is necessary, since you can already do stuff like right click the page background in a unstashable tab and choose to stash it so it seems like Tab Stash will let you stash "unstashable" tabs if you _really_ want to.

The main advantage of this pull is it makes a huge difference in how response Tab Stash feels when opening new tabs. Many sites can take a second or two to reach a "stashable" state and currently the Tab Stash interface just doesn't react until then.

josh-berry reviewed (commented):
This generally looks good; I was playing with it a bit and I think it might also be good to filter unstashable URLs from `visibleChildren` here:

https://github.com/josh-berry/tab-stash/blob/3e93fb07b054743e36f6194a8a616f1438445c0b/src/stash-list/window.vue#L162-L168

Otherwise the Tab Stash UI will stash itself if it's open in a tab (and this might cause the operation to stop in the middle if the user is using the "close tabs immediately" option).

You're right that it's possible to stash unstashable things if you really try (e.g. drag-and-drop, explicit selection, etc.) and I think I agree it's not *really* worth the effort to block this (especially because I got complaints when I stopped people from stashing the new-tab page, to use as a separator...).

josh-berry commented:
Do you want me to go ahead and merge this one (with the aforementioned change applied) even though there are open comments on some of the earlier PRs?

KerfuffleV2 commented:
@josh-berry 

> Otherwise the Tab Stash UI will stash itself if it's open in a tab 

Good point! I added a commit that fixes this.

> Do you want me to go ahead and merge this one

Absolutely. Also rebased this to current master.

josh-berry commented on src/stash-list/window.vue:
```diff
@@ -161,9 +161,14 @@ export default defineComponent({
 
         async stash(ev: MouseEvent | KeyboardEvent) {this.attempt(async() => {
             if (this.visibleChildren.length === 0) return;
```
One minor thing I caught here (and fixed during the merge)--we actually want to check `stashable_visible_children` directly, or we might still wind up creating an empty group (`createStashFolder()` below) even if there's nothing to stash.

Anyway, I took care of it, but wanted to mention in case you noticed the change during subsequent merges/rebases/etc.

KerfuffleV2 replied:
Oh, sorry. Thanks for noticing and fixing that issue!
